api/src/main/java/dev/jsinco/brewery/math/Range.java

```diff
@@ -0,0 +1,39 @@
+package dev.jsinco.brewery.math;
+
+import java.util.Random;
+
+public abstract class Range<T extends Number> {
+    protected static final Random RANDOM = new Random();
+
+    private T min;
+    private T max;
+
+    public Range(T min, T max) {
+        this.min = min;
```

Comment:
Not null checks/annotations might be a good idea to add here and to all child constructors